logstash-core/lib/logstash/pipeline.rb

```diff
@@ -501,30 +501,21 @@ def inputworker(plugin)
 
   # initiate the pipeline shutdown sequence
   # this method is intended to be called from outside the pipeline thread
-  # @param before_stop [Proc] code block called before performing stop operation on input plugins
-  def shutdown(&before_stop)
+  def shutdown
```

Comment:
Same as above for parity

```suggestion
  # and will block until the pipeline has successfully shut down.
  def shutdown
```